scipy/optimize/least_squares.py

```diff
@@ -0,0 +1,819 @@
+              Additionally  ``method='trf'`` supports  'regularize' option
+              (bool, default is True) which adds a regularization term to the
+              normal equations, which improves convergence if Jacobian is
+              rank-deficient [Byrd]_ (eq. 3.4).
+
+    jac_sparsity : {None, array_like, sparse matrix}, optional
+        Defines the sparsity structure of the Jacobian matrix for finite
+        differences. If the Jacobian has only few non-zeros in *each* row,
+        providing the sparsity structure will greatly speed up the computations
+        [Curtis]_. Should have shape (m, n). A zero entry means that a
+        corresponding element in the Jacobian is identically zero. If provided,
+        forces the use of 'lsmr' trust-region solver. If None (default) then
+        dense differencing will be used. Has no effect for 'lm' method.
+    verbose : {0, 1, 2}, optional
+        Level of algorithm's verbosity:
+
+            * 0 (default) - work silently.
+            * 1 - display a termination report.
+            * 2 - display progress during iterations (not supported by 'lm'
+              method).
+
+    args, kwargs : tuple and dict, optional
+        Additional arguments passed to `fun` and `jac`. Both empty by default.
+        The calling signature is ``fun(x, *args, **kwargs)`` and the same for
+        `jac`.
+
+    Returns
+    -------
+    `OptimizeResult` with the following fields defined:
+    x : ndarray, shape (n,)
+        Solution found.
+    cost : float
+        Value of the cost function at the solution.
+    fun : ndarray, shape (m,)
+        Vector of residuals at the solution.
+    jac : ndarray, sparse matrix or LinearOperator, shape (m, n)
+        Modified Jacobian matrix at the solution, in the sense that J^T J
+        is a Gauss-Newton approximation of the Hessian of the cost function.
+        The type is the same as was used by the algorithm.
+    grad : ndarray, shape (m,)
+        Gradient of the cost function at the solution.
+    optimality : float
+        First-order optimality measure. In unconstrained problems it is always
+        the uniform norm of the gradient. In constrained problems it is the
+        quantity which was compared with `gtol` during iterations.
+    active_mask : ndarray of int, shape (n,)
+        Each component shows whether a corresponding constraint is active
+        (that is, whether a variable is at the bound):
+
+            *  0 - a constraint is not active.
+            * -1 - a lower bound is active.
+            *  1 - an upper bound is active.
+
+        Might be somewhat arbitrary for 'trf' method as it does strictly
+        feasible iterates and `active_mask` is determined within a tolerance
+        threshold.
+    nfev : int
+        Number of function evaluations done. Methods 'trf' and 'dogbox' do not
+        count function calls for numerical Jacobian approximation, as opposed
```

Comment:
All methods count function evaluations other than required for Jacobian evaluation. So all methods can return sensible numbers.